Add option to show markdown report on step summary

KrijnL opened the pull request:
The motivation for creating this PR is that the code generates a really nice Markdown summary and then never uses it.
This conflicts with what is mentioned in the README, but I assume that's due to changes that happened in `v2`.

In any case it can be nice to show the report on pull requests, so that you get to see the errors that are not in your PR as well. Can also be useful on pulls if there were commits directly on a branch, without going through PR steps (ideally that should never happen, of course, but I've seen enough cases where it does...)

PS. I ran a `npm run build`, but if there are any other steps I need to take to build the action, please let me know!

KrijnL committed:
Add option to show markdown report on step summary

ataylorme reviewed (changes requested):
@KrijnL thank you for the pull request! You are right that the markdown summary missing is a mistake and was left out when refactoring for v2

I appreciate you working to add it back. I would request a small change. The [check run API documentation](https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#update-a-check-run) shows `output.text` as a string.

If the option to display the markdown summary is `false` please pass an empty string rather than `undefined`. This also means the `text` parameter for `closeStatusCheck` can be required and not optional.

KrijnL commented:
Hi Andrew,

Thanks for taking a look at my PR and merging the other ones so quickly!
Your requested change makes total sense. I've synced my forked repo with yours, rebased the branch and implemented the change you requested.

I'm not entirely sure why the merge check is failing. If there is anything I need to change, do let me know!

Warm regards

ataylorme commented:
@KrijnL the tests is failing due to security reasons `GITHUB_TOKEN` has only read access from fork PRs and needs `write` to run the Action.

I will test it locally and then, if things look good, merge this in.
